larpandoracontent/LArThreeDReco/LArHitCreation/ThreeDHitCreationAlgorithm.cc

```diff
@@ -175,6 +214,183 @@ void ThreeDHitCreationAlgorithm::IterativeTreatment(ProtoHitVector &protoHitVect
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+void ThreeDHitCreationAlgorithm::Project3DHit(const ProtoHit &hit, const HitType view, ProtoHit &projectedHit)
+{
+    projectedHit.SetPosition3D(
+        LArGeometryHelper::ProjectPosition(this->GetPandora(), hit.GetPosition3D(), view),
+        hit.GetChi2()
+    );
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void ThreeDHitCreationAlgorithm::GetSetIntersection(RANSACHitVector &first, RANSACHitVector &second, RANSACHitVector &result)
+{
+    const auto compareFunction = [] (const RANSACHit &a, const RANSACHit &b) -> bool {
+        return a.GetProtoHit().GetPosition3D().GetX() < b.GetProtoHit().GetPosition3D().GetX();
+    };
+
+    std::sort(first.begin(), first.end(), compareFunction);
+    std::sort(second.begin(), second.end(), compareFunction);
+
+    std::set_intersection(
+        first.begin(), first.end(),
+        second.begin(), second.end(),
+        std::inserter(result, result.begin()),
+        compareFunction
+    );
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void ThreeDHitCreationAlgorithm::ConsolidatedMethod(const ParticleFlowObject *const pPfo, ProtoHitVectorMap &allProtoHitVectors,
+        ProtoHitVector &protoHitVector)
+{
+    if (allProtoHitVectors.size() == 0)
+        return;
+
+    const float DISTANCE_THRESHOLD = 0.05; // TODO: Move to config option.
+    const std::vector<HitType> views = {TPC_VIEW_U, TPC_VIEW_V, TPC_VIEW_W};
+
+    std::map<HitType, RANSACHitVector> goodHits;
+
+    const std::vector<std::string> toolsToAvoid = {"Tool0039", "Tool0043"}; // TODO: Config option?
```

Comment:
I think this will need to be configurable. Presumably will also need to think about mapping tool names onto the ids, as I assume those numbers are necessarily stable.